pkg/velero/server/args.go

```diff
@@ -39,6 +39,13 @@ func (a Args) StringArr(dpaFeatureFlags []string, logLevel string) ([]string, er
 	} else if boolptr.IsSetToFalse(a.DefaultVolumesToFsBackup) {
 		args = append(args, "--default-volumes-to-fs-backup=false")
 	}
+	if a.UploaderType != "" {
```

Comment:
Shouldn't we also return an error if `UploaderType` is empty?

Reply:
We can return error or ignore and do not add that as a flag to the server args (empty arg is ignored in most CLI's). The ignore is current behavior but I am happy to change it to an error state, thought this isn't really error as it's empty=not passed. 

In here I was following also what is on line 49, which is logLevel if it's empty string it's ignored, so I think if we change it here we also change it to any other string being used as an arg passed to the server?

In our CSV we ensure it's required and it's either kopia or restic (enum type), this file `args.go` as I understand is used for testing/dev only and it's not really used by operator.

Reply:
"ignore and do not add that as a flag to the server args" would be defaulting to whatever velero defaults to (kopia in 1.12).
Then again, since UploaderType is now required in the dpa config, I guess this will always be non-empy, since DPA creation itself fails if it's empty.

Reply:
I believe we are talking here about two different parts of OADP:

1. `uploaderType` in DPA is required, but *only* when `nodeAgent` is defined. We need to remember that `nodeAgent` is *optional*, which makes `uploaderType` also non-existing for the scenario where `nodeAgent` is missing.
2. The `if a.UploaderType != "" ` on line 42 relates to the server args, which are not really connected to DPA and are used to configure server for testing/development, which makes that `UploaderType` optional and if it's not there the string will be empty, so we should check here for an empty string case. See:
  https://github.com/openshift/oadp-operator/blob/65281decc7c532255b9e04b88eac7963f8cc57ba/pkg/velero/server/config.go#L85

Reply:
ok. So we don't actually expect users to pass UploaderType via server args, though, do we? It's required with `nodeAgent`, but if they're not defining node agent, then 1 of 2 things is true:
1) they're using restic, so OADP is setting `UploaderType` to `restic` in the velero command
2) they're not using node agent at all, so it's irrelevant

So there's really no situation in which a user should be specifying UploaderTypevia server config, is there?

Reply:
you are right, I think the implementation needs to include that and pass `--use-node-agent` if `--uploader-type` is specified.

Should we have an option where `--use-node-agent` but *without* `--uploader-type` in upstream it is possible, then it defaults to the default `--uploader-type` but I am not convinced we should allow that.

My proposal is to inject  `--use-node-agent` whenever `uploaderType` is set in our server args?

Reply:
I'm not sure we want it there at all. If they want to enable node agent, they should use the node agent config. Otherwise we end up having to duplicate all of this logic under server args too. But also, --use-node-agent isn't a server arg. Using node agent doesn't add an arg to the server, it creates a separate DaemonSet to create node agent pods. So I think we can leave UploaderType out of our server args package completely.

Reply:
Removed that part as requested.